python-package/lightgbm/dask.py

```diff
@@ -136,25 +136,24 @@ def _train_part(params, model_factory, list_of_parts, worker_address_to_port, re
     is_ranker = issubclass(model_factory, LGBMRanker)
 
     # Concatenate many parts into one
-    parts = tuple(zip(*list_of_parts))
-    data = _concat(parts[0])
-    label = _concat(parts[1])
+    data = _concat([x['data'] for x in list_of_parts])
+    label = _concat([x['label'] for x in list_of_parts])
+
+    if 'weight' in list_of_parts[0]:
+        weight = _concat([x['weight'] for x in list_of_parts])
+    else:
+        weight = None
+
+    if 'group' in list_of_parts[0]:
+        group = _concat([x['group'] for x in list_of_parts])
+    else:
+        group = None
 
     try:
         model = model_factory(**params)
-
         if is_ranker:
-            group = _concat(parts[-1])
-            if len(parts) == 4:
-                weight = _concat(parts[2])
-            else:
-                weight = None
             model.fit(data, y=label, sample_weight=weight, group=group, **kwargs)
```

Comment:
I believe this whole `try` context can be simplified to single call of `model.fit(data, y=label, sample_weight=weight, group=group, **kwargs)`.
`group` param can be passed to the base `LGBMModel` class without the need to check whether it is ranker or not (`is_ranker`).

Reply:
Yep this is true, good catch. This was a holdover from the world before `fit` was estimator-type-agnostic and each Dask model subclass had its own fit method. Will change!

Reply:
Wait no - sorry I take prior comment back. This is a call to the *local*/serial model `fit` method. Classifier/Regressor don't accept `type` argument: ```Exception: TypeError("fit() got an unexpected keyword argument 'group'")```. So I don't think that `group` can be safely passed as a kwarg to all model types.

Reply:
Oh sorry, indeed! I forgot about that `LGBMClassifier` and `LGBMRegressor` overwrite `fit` method and `group` is not available there.